haystack/core/pipeline/pipeline.py

```diff
@@ -64,6 +64,37 @@ def __eq__(self, other) -> bool:
             return False
         return self.to_dict() == other.to_dict()
 
+    def __repr__(self) -> str:
+        """
+        Returns a text representation of the Pipeline.
+        If this runs in a Jupyter notebook, it will instead display the Pipeline image.
+        """
+        res = ""
+        try:
+            # We call draw here so that if we're in a Jupyter notebook we can embed the Pipeline image.
+            # If it fails it's no a big deal, we just don't show the image. Also it probably means we're not in a
+            # Jupyter notebook.
```

Comment:
We should add a utility function ([like so](https://github.com/explosion/spaCy/blob/e1249d3722765aaca56f538e830add7014d20e2a/spacy/util.py#L1079)) to detect Jupyter notebooks and use that instead of exception handling for control flow.